### PR TITLE
feat: add timeout parameter to polling while loops

### PR DESCRIPTION
**Pull request by bjhardcastle**

### CODE REVIEW

File: src/codeocean/computation.py [hunk 1]
```diff
@@ -141,20 +141,37 @@ def run_capsule(self, run_params: RunParams) -> Computation:
 
         return Computation.from_dict(res.json())
 
-    def wait_until_completed(self, computation: Computation, polling_interval: int = 5) -> Computation:
+    def wait_until_completed(
+        self, computation: Computation, polling_interval: float = 5, timeout: float | None = None,
```
Comment (zvikagart) [inline src/codeocean/computation.py:145]: To be consistent with argument formatting elsewhere in the package - if not in a single line then arg per line:
```suggestion
        self,
        computation: Computation,
        polling_interval: float = 5,
        timeout: Optional[float] = None,
```

File: src/codeocean/computation.py [hunk 2]
```diff
@@ -141,20 +141,37 @@ def run_capsule(self, run_params: RunParams) -> Computation:
...
+    ) -> Computation:
         """
-        Polls the given computation until it reaches the 'Completed' or 'Failed' state.
+        Polls the given computation until it reaches the 'Completed' or 'Failed'
+        state.
```
Comment (zvikagart) [inline src/codeocean/computation.py:149]: We allow max line length 120 so this isn't required.

File: src/codeocean/data_asset.py [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+from time import sleep, time
 from typing import Optional
 
-from codeocean.components import SortOrder, SearchFilter, Permissions
-from codeocean.computation import PipelineProcess, Param
+from dataclasses_json import dataclass_json
+from requests_toolbelt.sessions import BaseUrlSession
+
+from codeocean.components import Permissions, SearchFilter, SortOrder
+from codeocean.computation import Param, PipelineProcess
```
Comment (zvikagart) [inline src/codeocean/data_asset.py:11]: To be consistent with the import ordering in the rest of the package let's keep the grouping as before

File: src/codeocean/computation.py (hunk 1 above)
Comment (zvikagart) [reply src/codeocean/computation.py:145]: Let's also please apply the same in the rest of the PR

**Review by zvikagart — CHANGES_REQUESTED** [review]

File: src/codeocean/data_asset.py (hunk 3 above)
Comment (bjhardcastle) [reply src/codeocean/data_asset.py:11]: Bah I didn't notice isort ran in the IDE. Thanks

File: src/codeocean/data_asset.py [hunk 4]
```diff
@@ -268,7 +291,9 @@ def archive_data_asset(self, data_asset_id: str, archive: bool):
             params={"archive": archive},
         )
 
-    def search_data_assets(self, search_params: DataAssetSearchParams) -> DataAssetSearchResults:
+    def search_data_assets(
+        self, search_params: DataAssetSearchParams
+    ) -> DataAssetSearchResults:
```
Comment (zvikagart) [inline src/codeocean/data_asset.py:296]: Let's revert this change as it's not necessary

File: src/codeocean/data_asset.py [hunk 5]
```diff
@@ -227,8 +227,12 @@ def get_data_asset(self, data_asset_id: str) -> DataAsset:
...
         return DataAsset.from_dict(res.json())
 
-    def update_metadata(self, data_asset_id: str, update_params: DataAssetUpdateParams) -> DataAsset:
-        res = self.client.put(f"data_assets/{data_asset_id}", json=update_params.to_dict())
+    def update_metadata(
+        self, data_asset_id: str, update_params: DataAssetUpdateParams
+    ) -> DataAsset:
+        res = self.client.put(
+            f"data_assets/{data_asset_id}", json=update_params.to_dict()
+        )
```
Comment (zvikagart) [inline src/codeocean/data_asset.py:235]: Let's revert this change as it's not necessary
